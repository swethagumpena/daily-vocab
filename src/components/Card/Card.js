import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.scss';

const Card = ({ word, sentence }) => (
  <div>
    <div className={styles.card}>
      {/* <div>{JSON.stringify(vocab)}</div> */}
      <p className={styles.word}>{word}</p>
      <p className={styles.sentence}>{sentence}</p>
    </div>

  </div>
);

Card.propTypes = {
  word: PropTypes.string.isRequired,
  sentence: PropTypes.string.isRequired,
};

export default Card;
