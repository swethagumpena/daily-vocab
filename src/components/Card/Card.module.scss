@import "../../styles/index.scss";

.card {
    margin: 0.5rem;
    border-radius: 1.25rem;
    box-shadow: 0rem 0rem 0.25rem #9a9a9a;
    // width: calc((100% - (8 * 0.125rem + 8 * 0.5rem)) / 4);
    width: 21rem;
    display: flex;
    flex-direction: column;
    border: 0.125rem solid lightgrey;
    align-items: flex-start;
    background-color: $primary-background-color;
    color: $primary-text-color;
    font-family: "Inconsolata", monospace
}

.card p{
    margin: 1rem;
}

.word{
    font-weight: bold;
    font-size: 1.5rem;
}

// @media (max-width: 72rem) {
//     .card {
//         width:  calc((100% - (6 * 0.125rem + 6 * 0.5rem)) / 3);
//     }
// }

@media (max-width: 72rem) {
    .card {
        width:  20rem;
    }
}

// @media (max-width: 48rem) {
//     .card {
//         width:  calc((100% - (4 * 0.125rem + 4 * 0.5rem)) / 2);
//     }
// }

@media (max-width: 48rem) {
        .card {
            width:  22.75rem;
        }
    }

// @media (max-width: 30rem) {
//     .card {
//         width: 100%;
//     }
// }
