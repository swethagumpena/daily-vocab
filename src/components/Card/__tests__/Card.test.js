import { render, screen } from '@testing-library/react';
import React from 'react';
import Card from '../Card';

describe('example', () => {
  test('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
});

describe(Card.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<Card word="abet" sentence="abet a suicide" />);
    expect(container).toMatchSnapshot();
  });

  test('should display a word', () => {
    render(<Card word="abet" sentence="abet a suicide" />);
    expect(screen.getByText('abet'));
  });

  test('should display a sentence', () => {
    render(<Card word="abet" sentence="abet a suicide" />);
    expect(screen.getByText('abet a suicide'));
  });
});
