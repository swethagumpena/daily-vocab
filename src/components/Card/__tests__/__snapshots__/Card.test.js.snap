// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    data-testid="Card"
  >
    <div
      class="card"
    >
      <p
        class="word"
      >
        abet
      </p>
      <p
        class="sentence"
      >
        abet a suicide
      </p>
    </div>
  </div>
</div>
`;
